test/lib/MapTest.v3

```diff
@@ -0,0 +1,125 @@
+// Copyright 2023 Virgil authors. All rights reserved.
+// See LICENSE for details of Apache 2.0 license.
+
+def T = LibTests.register("Map", _, _);
+def X = [
+	T("get", test_get),
+	T("set", test_set),
+	T("has", test_has),
+	T("apply", test_apply),
+	T("remove", test_remove),
+	()
+];
+
+class Counter {
+	var x = 0;
+	def add<K, V>(a: K, b: V) { x++; }
+	def reset() { x = 0; } 
+}
+
+def int_map<V>() -> HashMap<int, V> {
+	return HashMap<int, V>.new(int.!, int.==);
+}
+
+def int2str(v: int) -> string {
+	return StringBuilder.new().putd(v).extract();
+}
+
+def size<K, V>(m: HashMap<K, V>) -> int {
+	var c = Counter.new();
+	m.apply(c.add);
+	return c.x;
+}
+
+def test_get(t: LibTest) {
+	var c = Counter.new();
+	var i_s = int_map<string>();
+	var s_s = Strings.newMap<string>();
+	for (i < 100) {
+		var s = int2str(i);
+		i_s[i] = s;
+		s_s[s] = s;
+	}
+	for (i < 100) {
```

Comment:
You might want to test the strings by identity, e.g. by keeping an array of pairs here.